conda_smithy/lint_recipe.py

```diff
@@ -149,6 +149,22 @@ def lintify(meta, recipe_dir=None):
     except RuntimeError as e:
         lints.append(str(e))
 
+    # 13: If cython is a build-dependency then python should be a run-time
+    # dependency. This ensures that packages will be marked correctly by conda
+    if 'cython' in requirements_section.get('build', '') and 'python' not in requirements_section.get('run', ''):
+        lints.append("When using cython then python should be a run-time dependency")
```

Comment:
This seems very specific. There are number of errors like this, but we can't add all of them. (swig, numpy as build dependency.)

How about something more generic like if python is an implicit runtime requirement, then it should be added explicitly in runtime requirements?